### ISSUE TITLE
New MacOS13 image is slower than previous one.

### ISSUE THREAD

**Issue by christiangal-indi** [actions/runner-images#8434]
### Description

Using the same Xcode 14.3.1 and the same code on the 2 different images of MacOs13 results in very different compile and test run times.

Image Version: 20230417.1
Complete pipeline 25-33 min

Image Version: 20230924.1
Complete pipeline 45-60(timeout) min

Here you can see the difference from "Wednesday"
![Screenshot 2023-10-02 at 15 49 40](https://github.com/actions/runner-images/assets/110092881/ba9e5fc1-c08d-44e9-81e9-a5b262837286)

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [X] macOS 13
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Agent name: 'Azure Pipelines 10'
Agent machine name: 'Mac-1696247891010'
Current agent version: '3.225.0'
Operating System
macOS
13.6
22G120
Runner Image
Image: macos-13
Version: 20230924.1
Included Software: https://github.com/actions/runner-images/blob/macos-13/20230924.1/images/macos/macos-13-Readme.md
Image Release: https://github.com/actions/runner-images/releases/tag/macos-13%2F20230924.1

### Is it regression?

Yes, Image Version: 20230417.1

### Expected behavior

The run should take around 30 min.

### Actual behavior

The run takes more than 40 min, and sometimes, fail with a 60min timeout.

### Repro steps

Compile a IOS app using Xcode.

**Comment by amseddi** [comment]
Looks similar to #8023.
Did you try specifying the run target to iOS 16?

**Comment by Alexey-Ayupov** [comment]
Hello, @christiangal-indi. Could you please provide minimal repro steps?

**Comment by christiangal-indi** [comment]
> Looks similar to #8023. Did you try specifying the run target to iOS 16 simulator?

Yes, those times are using the same simulator with version (16.4). That's why I have created another issue as it is not the same case. This is about the images itself and not the simulator that was selected.

**Comment by christiangal-indi** [comment]
> Hello, @christiangal-indi. Could you please provide minimal repro steps?
I have updated the description.

**Comment by andriyslyusar** [comment]
We can see very similar result in time in our project. Average time to builds and run unit test increase from 25 to 60 minutes.

**Comment by luis-ifolor** [comment]
We're also seeing a huge increase of execution time in our project just by switching to this image and trying to run the tests against iOS 17. Tests timeouts are happening where they previously didn't and the simulator process is also sometimes timing out.

**Comment by honkmaster** [comment]
We've noted the same: tests often fail to start, potentially linked to issue #8472. When they run, they're at least 25% slower. But if we're being honest: Isn't this more of a problem with Xcode than with the runner images?

**Comment by christiangal-indi** [comment]
> We've noted the same: tests often fail to start, potentially linked to issue #8472. When they run, they're at least 25% slower. But if we're being honest: Isn't this more of a problem with Xcode than with the runner images?

Not sure, it is the same xcode running in 2 different runner images. Something has changed between the 2 images.

**Comment by gredman** [comment]
Same, but I am hoping it’s caused by this issue, fixed in 15.1:

> Fixed: Executing Unit/UI tests from Xcode on the iOS Simulator takes an extended time to launch on first run of the suite. (115187363) (110330776)

**Comment by mluisbrown** [comment]
> Running the CI test suite locally )on an M1 Pro MacBook Pro) with Xcode 15 and iOS 17 is not noticeably slower than with Xcode 14.3.1 and iOS 16.4.

I was wrong about that, just assuming that 6-7 minutes to run all tests was "normal" on my machine. Turns out that with Xcode 15.0.1 and iOS 17.0.1 simulators tests run in 2-3 minutes, so I was seeing the issue locally too, it just wasn't as obvious.

**Comment by TomaszLizer** [comment]
> We are also seeing a 2-3x increase in CI time with `macos-13` and Xcode 15 running tests with iOS 17 simulators. I've also tried the `macos-13-xl` runner and it is equally slow.
> 
> Using the same `macos-13` runner with Xcode 14.3.1 and iOS 16.4 simulators was fine. Running the CI test suite locally )on an M1 Pro MacBook Pro) with Xcode 15 and iOS 17 is not noticeably slower than with Xcode 14.3.1 and iOS 16.4.
> 
> This is blocking our ability to move to using Xcode 15.

@mluisbrown Quite similar on my side: macos-13 runner with Xcode 14.3 - build and tests takes ~35-40 minutes.
Switching to Xcode 15 ends in timeout after 90 minutes and based on logs Fastlane has barely started building app (less than half of the way)

> Same, but I am hoping it’s caused by this issue, fixed in 15.0.1 (and 15.1):
> 
> > Fixed: Executing Unit/UI tests from Xcode on the iOS Simulator takes an extended time to launch on first run of the suite. (115187363) (110330776)

@gredman I fear not based on the above 😢

**Comment by mluisbrown** [comment]
@TomaszLizer our app is building fine, and in reasonable time, with Xcode 15. It's the testing that is killing it. There are times when there's a 15 minute wait whilst it appears to be doing nothing at all!

**Comment by TomaszLizer** [comment]
Oh, interesting 🤔 
Fingers crossed then 😃

**Comment by honkmaster** [comment]
Xcode 15.0.1 RC is published.

https://developer.apple.com/documentation/xcode-release-notes/xcode-15_0_1-release-notes

**Comment by christiangal-indi** [comment]
> > We are also seeing a 2-3x increase in CI time with `macos-13` and Xcode 15 running tests with iOS 17 simulators. I've also tried the `macos-13-xl` runner and it is equally slow.
> > Using the same `macos-13` runner with Xcode 14.3.1 and iOS 16.4 simulators was fine. Running the CI test suite locally )on an M1 Pro MacBook Pro) with Xcode 15 and iOS 17 is not noticeably slower than with Xcode 14.3.1 and iOS 16.4.
> > This is blocking our ability to move to using Xcode 15.
> 
> @mluisbrown Quite similar on my side: macos-13 runner with Xcode 14.3 - build and tests takes ~35-40 minutes. Switching to Xcode 15 ends in timeout after 90 minutes and based on logs Fastlane has barely started building app (less than half of the way)
> 
> > Same, but I am hoping it’s caused by this issue, fixed in 15.0.1 (and 15.1):
> > > Fixed: Executing Unit/UI tests from Xcode on the iOS Simulator takes an extended time to launch on first run of the suite. (115187363) (110330776)
> 
> @gredman I fear not based on the above 😢

We have seen problems in testing with the combination of Xcode15 and Fastlane, also on local. We were getting error 65 on tests execution and taking fastlane out solved the problem. So fastlane is the one that need to take action here. Test that on your side.

**Comment by mluisbrown** [comment]
@christiangal-indi I seriously doubt that Fastlane is the problem here, and have seen no evidence to support that. Xcode 15 has a known bug which causes this, which is fixed in Xcode 15.0.1 (and 15.1). Once 15.0.1 is released and made available on a GH runner that will be the first thing to test to see if/how things improve. I would only consider looking at other causes if that doesn't help.

**Comment by svenmuennich** [comment]
>Using the same macos-13 runner with Xcode 14.3.1 and iOS 16.4 simulators was fine. Running the CI test suite locally )on an M1 Pro MacBook Pro) with Xcode 15 and iOS 17 is not noticeably slower than with Xcode 14.3.1 and iOS 16.4.

We successfully use Xcode 15 on macOS 13 (M1) runners by _pairing_ them with iOS 16.4 simulators. They are not available by default, but creating them is easy, e.g.:
```sh
# Make iOS 16.4 simulator runtime available to Xcode 15
sudo mkdir -p /Library/Developer/CoreSimulator/Profiles/Runtimes
sudo ln -s /Applications/Xcode_14.3.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime /Library/Developer/CoreSimulator/Profiles/Runtimes/iOS\ 16.4.simruntime
# Create an iOS 16.4 simulator
xcrun simctl create "iPhone 14 Pro (16.4)" "com.apple.CoreSimulator.SimDeviceType.iPhone-14-Pro" "com.apple.CoreSimulator.SimRuntime.iOS-16-4"
```

With this combination (Xcode 15, iOS 16.4 simulator) we see similar test durations as with Xcode 14.

**Comment by mluisbrown** [comment]
The `macos-13` runner now includes Xcode 15.0.1 and unfortunately tests are still taking more than 2x as long as with Xcode 14.3.1 and iOS 16.x simulators.

**Comment by ilia-shipitsin** [comment]
@mluisbrown , do tests run fast on your local mac ? if so, we would like to do the same.
